.github/workflows/benchmark.yml

```
@@ -24,7 +24,15 @@ jobs:
         uses: actions/cache@v4
         with:
           path: ./benchmarks
-          key: ${{ runner.os }}-benchmark
+          # Use the current commit SHA as the cache key.
+          # This key won't exist on the first run, so the cache match falls back to restore-keys.
+          # Though, it won't be matched, the cache created will use this key as the cache key.
+          # So the next commit will be able to restore this cache (from the restore-keys).
+          # https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows#matching-a-cache-key
+          key: ${{ runner.os }}-benchmark-${{ github.sha }}
+          # `github.event.before` means the commit before the push (i.e. the previous commit).
+          # So we can fetch the exact benchmark data from the previous commit.
+          restore-keys: ${{ runner.os }}-benchmark-${{ github.event.before }}
```

Cannot we use the same pattern as here: https://github.com/open-telemetry/opentelemetry-go/pull/5160?

```suggestion
          key: ${{ runner.os }}-benchmark-${{ github.sha }}
          restore-keys: ${{ runner.os }}-benchmark-
```

---

If we use this pattern, we only match the most recent cache, which may not be the previous commit. For instance, a manually triggered benchmark on a PR commit will become the most recent cache, and the benchmark's result will be used for the next commit on the main branch.

We do not want this situation since we only want to see the comparison across the main branch.

```
restore-keys: ${{ runner.os }}-benchmark-
```

> If there are no exact matches, the action searches for partial matches of the restore keys. When the action finds a partial match, the most recent cache is restored to the path directory.

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows#cache-hits-and-misses

---

So, I prefer the current implementation with specific sha on `restore-keys`

---

Should we then do the same in https://github.com/open-telemetry/opentelemetry-go/blob/main/.github/workflows/links-fail-fast.yml if it will work as expected?

---

We don't need to update https://github.com/open-telemetry/opentelemetry-go/blob/main/.github/workflows/links-fail-fast.yml, as it is OK to use cache from non-main branches.